### ISSUE TITLE
Unknown Issue (Code 48?)

### ISSUE THREAD

**Issue by airsoftsoldrecn9** [andreafabrizi/Dropbox-Uploader#399]
It would appear curl is throwing CODE 48.
CURL_UNKOWN_OPTION (48) - An option passed to libcurl is not recognized/known. Refer to the appropriate documentation. This is most likely a problem in the program that uses libcurl. The error buffer might contain more specific information about which exact option it concerns.

A tad lost on how to troubleshoot this one. Any suggestions would be fantastic! ( BTW, I scrambled my Auth Code but I am using the correct one and my apologies on the code block formatting, all of the +'s were converted to bullet points. If I need to convert something, let me know).

<code>root@HomeServer:/tmp/home/root# dropbox_uploader space
1.0
Linux HomeServer 2.6.22.19 #7 Tue Nov 30 14:44:12 EST 2010 mips GNU/Linux
+ RESPONSE_FILE=/opt/share/dropbox_sync//du_resp_debug
+ [[ '' == '' ]]
+ BIN_DEPS='sed basename date grep stat dd mkdir curl'
+ CURL_BIN=curl
+ which sed basename date grep stat dd mkdir curl
+ [[ 0 != 0 ]]
+ which readlink
+ [[ 1 == 0 ]]
+ HAVE_READLINK=0
+ builtin printf ''
+ [[ 0 == 0 ]]
+ PRINTF='builtin printf'
+ PRINTF_OPT='-v o'
+ [[ -e /opt/etc/.dropbox_uploader ]]
+ source /opt/etc/.dropbox_uploader
+ [[ '' != '' ]]
+ [[ '' != '' ]]
+ [[ xxxxxxxxseCEZKLSFENSFELNGHYk8fOH00234oinshydsfelighn080K-iYUNKen_ = '' ]]
+ COMMAND=space
+ ARG1=
+ ARG2=
+ let argnum=1-1
+ case $COMMAND in
+ db_account_space
+ print 'Dropbox Uploader v1.0\n\n'
+ [[ 0 == 0 ]]
+ echo -ne 'Dropbox Uploader v1.0\n\n'
Dropbox Uploader v1.0

+ print ' > Getting space usage info... '
+ [[ 0 == 0 ]]
+ echo -ne ' > Getting space usage info... '
 > Getting space usage info... + curl -k -X POST -L -s --show-error --globoff -i -o /opt/share/dropbox_sync//du_resp_debug --header 'Authorization: Bearer xxxxxxxxseCEZKLSFENSFELNGHYk8fOH00234oinshydsfelighn080K-iYUNKen_' https://api.dropboxapi.com/2/users/get_space_usage
+ check_http_response
+ CODE=48
+ case $CODE in
+ grep -q 'HTTP/1.1 400' /opt/share/dropbox_sync//du_resp_debug
grep: /opt/share/dropbox_sync//du_resp_debug: No such file or directory
+ grep -q '^HTTP/1.1 200 OK' /opt/share/dropbox_sync//du_resp_debug
grep: /opt/share/dropbox_sync//du_resp_debug: No such file or directory
+ print 'FAILED\n'
+ [[ 0 == 0 ]]
+ echo -ne 'FAILED\n'
FAILED
+ ERROR_STATUS=1
+ remove_temp_files
+ [[ 1 == 0 ]]
+ exit 1
</code>

**Comment by chanio** [comment]
I only met with this error when the program was trying to renew my OAUTH-API1 code.
So, I just circumvented this part of the original script...
First, I obtained the new OAUTH-API2 as Dropbox help tells.
Then, in order for the program to find what it requires to keep on working as usual, I made a copy of the original dropbox-uploader.sh & original .dropbox-uploader , config file. 
And temporaly rewrote line 1468...
Instead of : 
OAUTH_ACCESS_TOKEN=$(sed -n 's/.*access_token": "\([^"]*\).*/\1/p' "$RESPONSE_FILE")
Just wrote: 
OAUTH_ACCESS_TOKEN="myNewAPI2OAUTHkey"
Then, when running the program, it modified the config file with just this new OAUTH-API2 key. Just what it was unsuccessfully wanting. 
Finally, overwrote my patched file with the original dropbox-uploader.sh.
And everything worked like a charm, again.
